app/goal_routes.py

```diff
@@ -0,0 +1,117 @@
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from .task_routes import validate_model
+from flask import Blueprint, jsonify, request, make_response, abort
+from datetime import datetime
+import os
+import requests
+goal_bp = Blueprint("goal_bp", __name__, url_prefix="/goals")
+
+# ----------------------------- ROUTES FOR GOAL MODEL -------------------------
+
+# create a new goal
+@goal_bp.route("",methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+
+    try:
+        new_goal = Goal.from_dict(request_body)
+    except:
+        return {"details": "Invalid data"}, 400
```

Comment:
Like we validated models, could we also make a general function that validates request bodies? Here's an example:
```python
def validate_request_body(request_body, keys):
    for key in keys:
        if not request_body.get(key): 
            abort(make_response({
                'Invalid Data': f'missing key: {key}'
            }, 400))

    return True
```
We can pass in the `request_body` and a list of strings that are `keys` and then check to see if those keys are present.